jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdConfiguration.java

```diff
@@ -59,11 +59,53 @@
      * @param clientSecret The client secret known only by the Client and the Authorization Server.
      */
     public OpenIdConfiguration(String provider, String clientId, String clientSecret)
+    {
+        this(provider, null, null, clientId, clientSecret);
+    }
+
+    /**
+     * Create an OpenID configuration for a specific OIDC provider.
+     * @param provider The URL of the OpenID provider.
+     * @param authorizationEndpoint the URL of the OpenID provider's authorization endpoint if configured.
+     * @param tokenEndpoint the URL of the OpenID provider's token endpoint if configured.
+     * @param clientId OAuth 2.0 Client Identifier valid at the Authorization Server.
+     * @param clientSecret The client secret known only by the Client and the Authorization Server.
+     */
+    public OpenIdConfiguration(String provider, String authorizationEndpoint, String tokenEndpoint, String clientId, String clientSecret)
     {
         this.openIdProvider = provider;
         this.clientId = clientId;
         this.clientSecret = clientSecret;
 
+        if (provider == null)
+            throw new IllegalArgumentException("Provider was not configured");
+
+        if (tokenEndpoint == null || authorizationEndpoint == null)
+        {
+            Map<String, Object> discoveryDocument = fetchOpenIdConnectMetadata(provider);
+
+            this.authEndpoint = (String)discoveryDocument.get("authorization_endpoint");
+            if (this.authEndpoint == null)
+                throw new IllegalArgumentException("authorization_endpoint");
+
+            this.tokenEndpoint = (String)discoveryDocument.get("token_endpoint");
+            if (this.tokenEndpoint == null)
+                throw new IllegalArgumentException("token_endpoint");
+
+            if (!Objects.equals(discoveryDocument.get("issuer"), provider))
+                LOG.warn("The provider in the metadata is not correct.");
+        }
+        else
+        {
+            this.authEndpoint = authorizationEndpoint;
+            this.tokenEndpoint = tokenEndpoint;
+        }
+
+        issuer = provider;
```

Comment:
One of these variables should be removed, no point having both when they are the same thing.

Reply:
I made this change, but it brought a few API changes. Can you please double check that in particular?